pkg/client/clientcmd/loader.go

```diff
@@ -134,8 +134,8 @@ func (rules *ClientConfigLoadingRules) Load() (*clientcmdapi.Config, error) {
 	// since values are overwritten, but maps values are not, we can merge the non-map config on top of the map config and
 	// get the values we expect.
 	config := clientcmdapi.NewConfig()
-	mergo.Merge(config, mapConfig)
-	mergo.Merge(config, nonMapConfig)
+	mergo.MergeWithOverwrite(config, mapConfig)
```

Comment:
This isn't completely accurate - previously, values would be stomped if a new non-zero value was set in object being merged. That this changed to no overwrite is what caused https://github.com/GoogleCloudPlatform/kubernetes/issues/12272.

I can look at this specific code in a bit but the intention here was just to maintain existing functionality.